operator/internal/controller/redpanda/redpanda_controller.go

```diff
@@ -408,24 +413,13 @@ func (r *RedpandaReconciler) reconcileDefluxed(ctx context.Context, rp *v1alpha2
 	return nil
 }
 
-func (r *RedpandaReconciler) reconcileClusterConfig(ctx context.Context, rp *v1alpha2.Redpanda) error {
+func (r *RedpandaReconciler) ratelimitCondition(ctx context.Context, rp *v1alpha2.Redpanda, conditionType string) bool {
```

Comment:
nit: add a comment here noting that this function require the Ready condition to be true.

`shouldReconcileCondition` might be a bit more descriptive and ergonomic?